src/core/lib/channel/context.h

```diff
@@ -68,10 +71,35 @@ struct grpc_call_context_element {
 };
 
 namespace grpc_core {
+class Call;
+
 // Bind the legacy context array into the new style structure
 // TODO(ctiller): remove as we migrate these contexts to the new system.
 template <>
 struct ContextType<grpc_call_context_element> {};
+
+// Also as a transition step allow exposing a GetContext<T> that can peek into
+// the legacy context array.
+namespace promise_detail {
+template <typename T>
+struct OldStyleContext;
```

Comment:
Is this supposed to be part of this PR?  I don't see it actually being used anywhere.

Reply:
Yeah... it's leveraged every time we do `GetContext<Call>()` to first grab the legacy context, and then look up the context we need.